packages/big-design/src/utils/useComponentSize.ts

```diff
@@ -0,0 +1,48 @@
+import { useCallback, useEffect, useState, RefObject } from 'react';
+
+interface ComponentSize {
+  height: HTMLElement['offsetHeight'];
+  width: HTMLElement['offsetWidth'];
+}
+
+const INITIAL_STATE: ComponentSize = {
+  height: 0,
+  width: 0,
+};
+
+const getSize = <T extends any>(element: RefObject<T>['current']): ComponentSize => {
+  if (element) {
```

Comment:
🍹 ternary

Reply:
Would it make sense to return a copy of `INITIAL_STATE`? Since it's going to be shared among all instances.